wagtail/snippets/tests/test_snippets.py

```diff
@@ -1641,6 +1692,122 @@ def test_get_after_publish_and_save_draft(self):
         )
 
 
+class TestSnippetUnpublish(TestCase, WagtailTestUtils):
+    def setUp(self):
+        self.user = self.login()
+        self.snippet = DraftStateModel.objects.create(text="to be unpublished")
+        self.unpublish_url = reverse(
+            "wagtailsnippets_tests_draftstatemodel:unpublish",
+            args=(quote(self.snippet.pk),),
+        )
+
+    def test_unpublish_view(self):
+        """
+        This tests that the unpublish view responds with an unpublish confirm page
+        """
+        # Get unpublish page
+        response = self.client.get(self.unpublish_url)
+
+        # Check that the user received an unpublish confirm page
+        self.assertEqual(response.status_code, 200)
+        self.assertTemplateUsed(response, "wagtailadmin/shared/confirm_unpublish.html")
+
+    def test_unpublish_view_invalid_pk(self):
+        """
+        This tests that the unpublish view returns an error if the object pk is invalid
+        """
+        # Get unpublish page
+        response = self.client.get(
+            reverse(
+                "wagtailsnippets_tests_draftstatemodel:unpublish", args=(quote(12345),)
+            )
+        )
+
+        # Check that the user received a 404 response
+        self.assertEqual(response.status_code, 404)
+
+    def test_unpublish_view_bad_permissions(self):
+        """
+        This tests that the unpublish view doesn't allow users without unpublish permissions
+        """
+        # Remove privileges from user
+        self.user.is_superuser = False
+        self.user.user_permissions.add(
+            Permission.objects.get(
+                content_type__app_label="wagtailadmin", codename="access_admin"
+            )
+        )
+        self.user.save()
+
+        # Get unpublish page
+        response = self.client.get(self.unpublish_url)
+
+        # Check that the user received a 302 redirected response
+        self.assertEqual(response.status_code, 302)
+
+    def test_unpublish_view_post(self):
+        """
+        This posts to the unpublish view and checks that the object was unpublished
+        """
+        # Connect a mock signal handler to unpublished signal
+        mock_handler = mock.MagicMock()
+        unpublished.connect(mock_handler)
+
+        # Post to the unpublish view
+        response = self.client.post(self.unpublish_url)
+
+        # Should be redirected to explorer page
+        self.assertRedirects(
+            response, reverse("wagtailsnippets_tests_draftstatemodel:list")
+        )
+
+        # Check that the page was unpublished
+        self.assertFalse(DraftStateModel.objects.get(pk=self.snippet.pk).live)
+
+        # Check that the page_unpublished signal was fired
```

Comment:
I thought it wouldn't be the `page_unpublished`?

Reply:
Oops yes, copy-paste leftover, it should just be `unpublished`.